infrastructure/code/sast_local_engine.go

```diff
@@ -0,0 +1,54 @@
+/*
+ * © 2023 Snyk Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package code
+
+import (
+	"github.com/rs/zerolog/log"
+	sglsp "github.com/sourcegraph/go-lsp"
+
+	"github.com/snyk/snyk-ls/application/config"
+	"github.com/snyk/snyk-ls/infrastructure/snyk_api"
+)
+
+const localCodeEngineWarning = "Warning! Local engine is enabled but URL is not configured."
+
+func (sc *Scanner) isLocalEngineEnabled(sastResponse snyk_api.SastResponse) bool {
+	log.Info().Any("sastResponse", sastResponse).Msg("sast response")
+	if sastResponse.SastEnabled && sastResponse.LocalCodeEngine.Enabled {
+		return true
+	}
+
+	return false
+}
+
+func (sc *Scanner) updateCodeApiLocalEngine(sastResponse snyk_api.SastResponse) bool {
+		method := "updateCodeApiLocalEngine"
+
+		if sc.isLocalEngineEnabled(sastResponse) && len(sastResponse.LocalCodeEngine.Url) > 1 {
+						config.CurrentConfig().SetSnykCodeApi(sastResponse.LocalCodeEngine.Url)
+						api := config.CurrentConfig().SnykCodeApi()
+						log.Info().Str("Updated SnykCodeApi to", api).Msg("SnykCodeApi Update - Local engine")
```

Comment:
In this case it could be: `log.Debug().Str("snykCodeApi", api).Msg("updated Snyk Code API Local Engine")`. Also, This information would be better to have as `Debug` level and the log is easier to read, if you add a `Str("method", method)`